### PR TITLE
Adding Some Actual Rules

### PR DESCRIPTION
**Pull request by codecaffeine**
These are pretty basic and just a start

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,3 +4,130 @@ TailoredSwiftStyleGuide
...
+    case Mercury, Venus, Earth, Mars, Jupiter, Saturn, Uranus, Neptune
+}
+```
+
+## Types
+- Arrays & dictionaries should be declared using the longhand form:
+
+```swift
+let people = [Person]      // ✘ Not This
+let people = Array<Person> // ✔ This
```
Comment (rmigneco) [inline README.md]: not really feeling this suggestion because Apple's Docs and examples typically use the shorthand form.

Comment (ndis1) [inline README.md]: image

File: README.md [hunk 2]
```diff
@@ -4,3 +4,130 @@ TailoredSwiftStyleGuide
...
+
+let properties = [String: AnyObject]           // ✘ Not This
```
Comment (rmigneco) [inline README.md]: same comment

File: README.md [hunk 3]
```diff
@@ -4,3 +4,130 @@ TailoredSwiftStyleGuide
...
+let properties = Dictionary<String, AnyObject> // ✔ This
+```
+
+- Optional should be declared using the shorthand form:
+
+```swift
+let person = Optional<Person> // ✘ Not This
```
Comment (rmigneco) [inline README.md]: while I agree with using shorthand for of optionals, I'm not onboard with using Longhand form for _some_ things (i.e. collections) and not others (i.e. optionals)

File: README.md [hunk 4]
```diff
@@ -4,3 +4,130 @@ TailoredSwiftStyleGuide
...
+- End files with a single newline.
+    - Set `CGG_WARN_ABOUT_MISSING_NEWLINE = YES` where applicable.
+- Don’t leave trailing whitespace.
+- Use 4 spaces, not a tab for indentation. This is the default in Xcode.
+
+## Naming
+- Always capitalize types including custom `typealias`ed type.
+- Always lowercase variables
+- When declaring variable, there should be no space between the name and the colon, but 1 space between the colon and the type
+
```
Comment (KVTaniguchi) [inline README.md]: emoji's in names yes / no?

File: README.md [hunk 5]
```diff
@@ -4,3 +4,130 @@ TailoredSwiftStyleGuide
...
+```swift
+// ✘ Not This
+dispatch_async(queue) { () -> Void in
+    print("Fired.")
+}
+
+// ✔ This
+dispatch_async(queue) {
+    print("Fired.")
```
Comment (ndis1) [inline README.md]: I don't want to fight autocomplete boo to this

File: README.md [hunk 6]
```diff
@@ -4,3 +4,130 @@ TailoredSwiftStyleGuide
...
+## Whitespace
+- Use vertical white space judiciously to organize units of code, especially within method bodies.
+- End files with a single newline.
+    - Set `CGG_WARN_ABOUT_MISSING_NEWLINE = YES` where applicable.
+- Don’t leave trailing whitespace.
+- Use 4 spaces, not a tab for indentation. This is the default in Xcode.
+
+## Naming
+- Always capitalize types including custom `typealias`ed type.
+- Always lowercase variables
```
Comment (ndis1) [inline README.md]: `camelcase`

File: README.md (hunk 4 above)
Comment (rmigneco) [inline README.md]: Nah

Comment (ndis1) [inline README.md]: I think they should be required in the examples in our pods

File: README.md [hunk 7]
```diff
@@ -4,3 +4,130 @@ TailoredSwiftStyleGuide
...
+enum Planet {
+    case Mercury, Venus, Earth, Mars, Jupiter, Saturn, Uranus, Neptune
+}
+```
+
+## Types
+- Arrays & dictionaries should be declared using the longhand form:
+
+```swift
+let people = [Person]      // ✘ Not This
```
Comment (KVTaniguchi) [inline README.md]: yes to this, its Apples recommended form, and I think the syntax here is wrong should be ```let people = [Person]()```

File: README.md (hunk 2 above)
Comment (KVTaniguchi) [inline README.md]: agree with @rmigneco  here, also syntax here is incorrect for declaring an empty array

File: README.md (hunk 4 above)
Comment (jgrandelli) [inline README.md]: @KVTaniguchi though they are super fun. And probably awesome to use in example projects, they should absolutely not be used in production code

File: README.md (hunk 1 above)
Comment (jgrandelli) [inline README.md]: fine

File: README.md (hunk 2 above)
Comment (jgrandelli) [inline README.md]: fine

File: README.md (hunk 5 above)
Comment (jgrandelli) [inline README.md]: 100% agree. There should be a caveat here for autocomplete.

Comment (codecaffeine) [inline README.md]: Should we make this be a non-rule? Follow if you want to?

**Comment by jgrandelli** [comment]
We should certainly add a section for access modifiers

File: README.md (hunk 5 above)
Comment (jgrandelli) [inline README.md]: eh i like it as a rule if you are declaring your own closures, but if you are letting Xcode fill it in and it does, i'd prefer to make it optional in those cases. However I will definitely comment on it if I know it wasn't autocomplete

File: README.md (hunk 6 above)
Comment (codecaffeine) [inline README.md]: fixed

**Comment by jgrandelli** [comment]
Do we also want to add a section around using things like map/reduce/filter where appropriate instead of traditional for loops?

**Comment by jgrandelli** [comment]
I know I am asking for a ton of extra, but if you want that $30 jharoka card you gotta earn it

**Comment by jgrandelli** [comment]
oh and same for unwrapping optionals

**Comment by codecaffeine** [comment]
I disambiguated the camel case with the upper/lower case.

**Pull request merged**